src/nodes/treeensembleclassifier.cc

```diff
@@ -0,0 +1,236 @@
+	}
+
+	node_tree_ids = parse_attribute_ints(nameToAttributeMap["nodes_treeids"]);
+	node_node_ids = parse_attribute_ints(nameToAttributeMap["nodes_nodeids"]);
+	leaf_tree_ids = parse_attribute_ints(nameToAttributeMap["class_treeids"]);
+	leaf_node_ids = parse_attribute_ints(nameToAttributeMap["class_nodeids"]);
+	leaf_class_ids = parse_attribute_ints(nameToAttributeMap["class_ids"]);
+	leaf_node_weights = parse_attribute_floats(nameToAttributeMap["class_weights"]);
+	node_true_ids = parse_attribute_ints(nameToAttributeMap["nodes_truenodeids"]);
+	node_false_ids = parse_attribute_ints(nameToAttributeMap["nodes_falsenodeids"]);
+	node_feature_ids = parse_attribute_ints(nameToAttributeMap["nodes_featureids"]);
+	node_thresholds = parse_attribute_floats(nameToAttributeMap["nodes_values"]);
+	node_modes = parse_attribute_strings(nameToAttributeMap["nodes_modes"]);
+	class_ids = parse_attribute_ints(nameToAttributeMap["classlabels_int64s"]);
+
+	// Check if there exists a class unused by a leaf
+	unused_class_id = -1; // Assume no class ID will be -1 (unconfirmed)
+	std::set<int64_t> class_ids_set(leaf_class_ids.begin(), leaf_class_ids.end());
+	for( int64_t class_id : class_ids )
+	{
+		// Assume only one class can be unused
+		if( !class_ids_set.contains(class_id) )
+		{
+			unused_class_id = class_id;
+			break;
+		}
+	}
+}
+
+
+/* Assign input tensors, resolve output tensor shapes, allocate output tensors */
+void TreeEnsembleClassifier::resolve(void)
+{
+	name_input(0, "features");
+
+	/* Create output tensors.
+	 * Set data dimensions and data type for the created tensors. */
+	Tensor *label_tensor = new Tensor;
+	label_tensor->data_dim.push_back(1);
+	label_tensor->data_type = onnx::TensorProto_DataType_INT64;
+
+	Tensor *probabilities_tensor = new Tensor;
+	probabilities_tensor->data_dim = {static_cast<int>(class_ids.size() + 1)};
+	probabilities_tensor->data_type = onnx::TensorProto_DataType_FLOAT;
+
+	// If outputs aren't registered in the same order
+	// as the model outputs are listed they will be reversed.
+	// Source: https://github.com/kraiskil/onnx2c/blob/855a0d7b86b1bb50aa45f0f78e4eefa6dc553085/src/graph.cc#L356
+	register_output(label_tensor, "label");
+	register_output(probabilities_tensor, "probabilities");
+}
+
+static std::string convertMode(std::string& mode)
+{
+	if( mode == "BRANCH_LEQ" )
+		return "<=";
+	if( mode == "BRANCH_LT" )
+		return "<";
+	if( mode == "BRANCH_GTE" )
+		return ">=";
+	if( mode == "BRANCH_GT" )
+		return ">";
+	if( mode == "BRANCH_EQ" )
+		return "==";
+	if( mode == "BRANCH_NEQ" )
+		return "!=";
+	if( mode == "LEAF" )
+		ERROR("Attempt to convert LEAF mode in TreeEnsembleClassifier");
+	ERROR("Unknown mode in TreeEnsembleClassifier");
+};
+
+std::unordered_map<int64_t, TreeEnsembleClassifier::Tree> TreeEnsembleClassifier::generateTreeMap() const
+{
+	std::unordered_map<int64_t, Tree> trees;
+
+	for( int i = 0; i < node_tree_ids.size(); ++i )
```

Comment:
```suggestion
	for( unsigned i = 0; i < node_tree_ids.size(); ++i )
```

Reply:
Good catches. I wonder why these didn't get flagged on my local builds. Maybe my compiler thinks these are fine or there is a CMake issue.

Reply:
I think I add `-Wall -Werror` to the Debug build only, but I'd guess you are building that one too. Maybe someone in the compilier project thought the warning is pretty silly (which it actually is, that should not cause problems as long as the tree is not insanely big). FWIW, I'm on gcc 9.4.

Mostly these are annoyance - I want to keep the build warning free just to not become unsensitized to warnings :)

Reply:
I'm on Clang 15. My `compile_commands.json` seems to have `-Wall -Werror` also. Very weird, I'd expect to see the same warning. Perhaps worth looking into.